lib/index.js

```diff
@@ -2,35 +2,40 @@
 
 const Boom = require('boom');
 const Hoek = require('hoek');
+const Joi = require('joi');
 
 // Declare Internals
 
 const internals = {};
 
-exports.register = (server, options, next) => {
-
-    server.auth.scheme('bearer-access-token', internals.implementation);
-    next();
+internals.defaults = {
+    accessTokenName: 'access_token',
+    allowQueryToken: true,
+    allowCookieToken: false,
+    allowMultipleHeaders: false,
+    allowChaining: false,
+    tokenType: 'Bearer'
 };
 
-exports.register.attributes = {
-    pkg: require('../package.json')
-};
+internals.schema = Joi.object().keys({
+    validateFunc: Joi.func().required(),
+    accessTokenName: Joi.string().required(),
+    allowQueryToken: Joi.boolean(),
+    allowCookieToken: Joi.boolean(),
+    allowMultipleHeaders: Joi.boolean(),
+    allowChaining: Joi.boolean(),
+    tokenType: Joi.string().required()
+});
 
 internals.implementation = (server, options) => {
 
     Hoek.assert(options, 'Missing bearer auth strategy options');
```

Comment:
I think it's fine as is, no need to use:
```
const settings = Hoek.applyToDefaults(internals.defaults, options || {});
```

So, what part of adding a test-case are you unsure about? This is why we're here to help :) When you're on the branch of this PR, any changes you make to any files and commit will be pushed to this PR.  

I'd like to add a test to make sure the error `Missing bearer auth strategy options` throws when valid options aren't provided if possible. If you look in `./test/index.js` you'll see examples of existing tests. 

Do you think you add a test that tries registering an auth strategy without providing any options, catch the error & check the error message is `Missing bearer auth strategy options`? If you're not interested in doing this part it's fine, but it's good learn how to write tests for the code!